app.py

```diff
@@ -1,120 +0,0 @@
-            except ValueError:
-                # Third option is invalid.
-                response_text = ("Sorry! `%s` isn't a valid sorting number. Try `/reddit %s %s 5`"
-                                 % (subreddit, sort_type, num_results))
-                return _create_response(response_text)
-        else:
-            # No option for results to return.
-            return fetch_posts(options[0], sort_type)
-    else:
-        if isinstance(options[1], int):
-            # For the case that someone passes `/reddit [subreddit] [sort_num]`
-            return fetch_posts(options[0], 'hot', options[1])
-        else:
-            return _create_response('Invalid search. Try `/reddit aww top 5` or `/reddit help`.')
-
-
-def _get_post_header(subreddit, sort, top_num):
-    """Formats the top heading to make it a bit more readable.
-    """
-    url = 'http://reddit.com/r/' + subreddit
-    if sort == 'new':
-        heading = '%s Newest Posts from' % (top_num)
-    elif sort == 'top':
-        heading = 'Top %s Posts from' % (top_num)
-    elif sort == 'rising':
-        heading = '%s Fastest Rising Posts' % (top_num)
-    else:
-        heading = '%s Hottest Posts' % (top_num)
-    heading += ' <%s|/r/%s>\n' % (url, subreddit)
-    return heading
-
-
-def fetch_posts(subreddit, sort='hot', top_num=5):
-    """Fetch post information from a subreddit given the options.
-    """
-    try:
-        sub = Reddit.get_subreddit(subreddit)
-        resp = getattr(sub, SORT_TYPES[sort])(limit=top_num)
-        data = _get_post_header(subreddit, sort, top_num)
-        for i, post in enumerate(resp):
-            post_number = i + 1
-            data += ('%s. *[%s]* <%s|%s>\n' %
-                     (post_number, post.score, post.url, post.title))
-        return _create_response(data)
-    except praw.errors.InvalidSubreddit:
-        return _create_response('Specify a subreddit that exists!')
-
-
-def help_option():
-    """Return text for `/reddit help`
-    """
-    help_text = """In order to use, you must specify a subreddit and can specify the type (hot [default], rising, new, or top), the number of results displayed or both! Some valid queries:
-    `/reddit aww 5`
-    `/reddit nba`
-    `/reddit oddlysatisfying new 4`
-Tweet @arjunblj if you have any other questions (or bugs) -- enjoy!"""
-    return Response(help_text, content_type='text/plain; charset=utf-8')
-
-
-@app.route('/search', methods=['post'])
-def search():
-    """i.e. /reddit nba rising 10
-    """
-    options = request.values.get('text').split(' ')
-    if not options[0]:
-        _create_response('You need to specify an option: try `/reddit aww top 5`.')
-    elif len(options) is 1:
-        if options[0] == 'help':
-            return help_option()
-        else:
-            return fetch_posts(options[0])
-    else:
-        try:
-            return _parse_options(options)
-        except:
-            return _create_response('Sorry, your request was incorrect.')
```

Comment:
I feel like this code would have made more sense had this been at the top, and then the more specific methods were at the bottom. (At least it would have made understanding `options` easier).

I know facundo likes the private functions before the public functions, but I'm starting to agree more with Clean Code, which suggests that the class functions get more specific the further down you travel. Just some food for thought - haven't finalized an opinion here yet.